Lib/test/test_zipimport.py

```diff
@@ -669,6 +669,26 @@ def testBytesPath(self):
 class CompressedZipImportTestCase(UncompressedZipImportTestCase):
     compression = ZIP_DEFLATED
 
+    @support.cpython_only
+    def test_issue31602(self):
+        # There shouldn't be an assertion failure in zipimporter.get_source()
+        # in case of a bad zlib.decompress().
+        import zlib
+        def bad_decompress(*args):
+            return None
+        z = ZipFile(TEMP_ZIP, 'w')
+        try:
+            name = 'bar.py'
+            data = b'spam'
+            z.writestr(name, data, ZIP_DEFLATED)
+            z.close()
+            zi = zipimport.zipimporter(TEMP_ZIP)
+            with support.swap_attr(zlib, 'decompress', bad_decompress):
+                self.assertRaises(TypeError, zi.get_source, 'bar')
+        finally:
+            z.close()
+            os.remove(TEMP_ZIP)
```

Comment:
You can use [`self.addCleanup()`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.addCleanup) so you don't have to specify this in the `try` block.